### DISCUSSION TITLE
How to open the project source code in Helix / Vscode

### DISCUSSION THREAD

**Discussion by thanh2k5vn** [diesel-rs/diesel#4004] [Q&A]
Hi @weiznich , thanks so much for a great library, I'm trying to learn about the source code of diesel, But when I clone the git repo and open it in vscode / helix, the rust-analyzer always complains about things like the attached image. Just wondering how I can fix this. Thanks so much 

<img width="1876" alt="image" src="https://github.com/diesel-rs/diesel/assets/145553981/8d544988-37f5-4ec5-a2f0-43aa70853bab">

**Comment by weiznich** [comment]
I neither use VSCode nor helix myself so I cannot comment in detail about these issues. What I would guess based on the error messages is that it tries to setup rust-analyzer with an incompatible list of features for the the specific crates. It's currently only possible to build tests if only a single feature backend feature is enabled, so enabling all backend features at once will cause inference issues for the test code.

**Comment by thanh2k5vn** [comment]
Hi @weiznich , can you share what editor you're using ?

**Reply by weiznich** [reply]
I'm using emacs locally, but I don't think it's important. 

As already written above I believe the following points are important:

* You need to configure rust-analyzer to use the right feature set and not just enable all features at once. 
* Even with that there are known bugs in rust-analyzer that might result in false positives be shown. Report them at the rust-analyzer repo if you find them, otherwise use `cargo check` or similar commands to locally ensure that the code is correct.

**Comment by thanh2k5vn** [comment]
ah, I see, I ran cargo check and it runs smoothly, then I realize that maybe there something wrong with RA, I check the log of helix and found this 

> 2024-05-02T14:58:41.602 helix_lsp::transport [ERROR] rust-analyzer err <- "2024-05-02T07:58:41.602537Z ERROR project_model::workspace: cyclic deps: diesel_derives(Idx::<CrateData>(14)) -> diesel(Idx::<CrateData>(11)), alternative path: diesel(Idx::<CrateData>(11)) -> diesel_derives(Idx::<CrateData>(14))\n"
2024-05-02T14:58:41.602 helix_lsp::transport [ERROR] rust-analyzer err <- "2024-05-02T07:58:41.602562Z ERROR project_model::workspace: cyclic deps: migrations_macros(Idx::<CrateData>(23)) -> diesel_migrations(Idx::<CrateData>(21)), alternative path: diesel_migrations(Idx::<CrateData>(21)) -> migrations_macros(Idx::<CrateData>(23))\n"


I found another similar issue of loco.rs  (https://github.com/loco-rs/loco/issues/490)

So pretty sure that it's not the issue with the project, Thanks so much @weiznich
